provisioner/clusterpy.go

```diff
@@ -324,9 +320,9 @@ func (p *clusterpyProvisioner) Provision(ctx context.Context, logger *log.Entry,
 			bucketName:     bucketName,
 			logger:         logger,
 			encodeUserData: true,
+			instanceTypes:  instanceTypes,
```

Comment:
I moved the `instanceTypes` field from `AWSNodePoolProvisioner` to here, where it is more generalized. This way the same field can be used in the render functions for `KarpenterNodePoolProvisioner` as well.